src/WebJobs.Script.WebHost/StandbyManager.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Description;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Config
+{
+    /// <summary>
+    /// Contains methods related to standby mode (placeholder) app initialization.
+    /// </summary>
+    public static class StandbyManager
+    {
+        private const string WarmUpFunctionName = "WarmUp";
+        private const string WarmUpAlternateRoute = "CSharpHttpWarmup";
+
+        public static async Task<HttpResponseMessage> WarmUp(HttpRequestMessage request, WebScriptHostManager scriptHostManager)
+        {
+            var queryParams = request.GetQueryParameterDictionary();
+            string value = null;
+            if (queryParams.TryGetValue("restart", out value) && string.Compare("1", value) == 0)
+            {
+                scriptHostManager.RestartHost();
```

Comment:
Allows the warmup request to cause a host restart if needed, to recycle things and exercise more of the code